### PR TITLE
Wearing Hailer gives unique death SFX

### PR DESCRIPTION
**Pull request by DrDiasyl**
## About The Pull Request

If you die with a SecHailer on, a unique sound effect will play (created entirely by me using in-game files):

https://github.com/tgstation/tgstation/assets/42353186/042e4159-57f3-4cc1-8d96-4dc7289f194b

(Please note that the pitch is randomized)
You can give any mask's unique death sound effects now.
## Why It's Good For The Game
SecHailers can change people's voices. Why not make them sound different when they die? It could be a humorous reference with a unique sound.
SecHailers' voice changer function is a reference, which I expanded by also making a unique death sound effect.

Also now you can make more masks with unique death sound effects if you want to.
## Changelog
:cl:
sound: Dying with a SecHailer on your face will make a unique death sound
/:cl:

### CODE REVIEW

File: code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -107,9 +107,14 @@
 		message_animal_or_basic = custom_message
 	. = ..()
 	message_animal_or_basic = initial(message_animal_or_basic)
+	var/mob/living/carbon/carbon_user = user
+	if(!user.can_speak() || user.oxyloss >= 50)
+		return //stop the sound if oxyloss too high/cant speak
+	if(istype(carbon_user)) // For masks that give unique death sounds
+		if(isclothing(carbon_user.wear_mask) && ((carbon_user.wear_mask.visor_flags & UNIQUE_DEATH_SOUND) || (carbon_user.wear_mask.clothing_flags & UNIQUE_DEATH_SOUND)))
+			playsound(carbon_user, carbon_user.wear_mask.unique_death, 200, TRUE, TRUE)
+			return
```
Comment (MrMelbert) [inline code/modules/mob/living/emote.dm:116]: The `UNIQUE_DEATH_SOUND` flag feels redundant, I think we should just check `if(mask.unique_death) play_unique_death_sound()`.

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/emote.dm (hunk 1 above)
Comment (DrDiasyl) [reply code/modules/mob/living/emote.dm:116]: > The `UNIQUE_DEATH_SOUND` flag feels redundant, I think we should just check `if(mask.unique_death) play_unique_death_sound()`.

Yeah fair enough. Any ideas how to fix the runtime error with monkey_business though?

**Comment by Fikou** [comment]
> SecHailers already change the voice for people, so why wouldn't it make them die with a different sound?

what

**Comment by DrDiasyl** [comment]
> > SecHailers already change the voice for people, so why wouldn't it make them die with a different sound?
> 
> what

SecHailers voice changer function literally makes them a Metrocop reference, so I expanded it

**Comment by Iamgoofball** [comment]
this is great, good work 👍

**Comment by nikothedude** [comment]
people will inevitably complain about sec powercreep since humans dont usually have a deathrattle but im of the opinion all species should have one

though, as i type i realize this IS actually a little sec powercreep because its an audible signal to everyone around that a secoff just died - even if humans deathrattled, it'd still be a very teeny buff

i dont think its worth really giving a shit about though

File: code/modules/mob/living/emote.dm [hunk 2]
```diff
@@ -107,9 +107,14 @@
 		message_animal_or_basic = custom_message
 	. = ..()
 	message_animal_or_basic = initial(message_animal_or_basic)
+	var/mob/living/carbon/carbon_user = user
+	if(!user.can_speak() || user.oxyloss >= 50)
```
Comment (nikothedude) [inline code/modules/mob/living/emote.dm:111]: are you sure there isnt a define for this? this is trying to mimick the behavior of going unconcious at 50+ oxyloss, right?

Comment (nikothedude) [reply code/modules/mob/living/emote.dm:111]: well not mimick, but use it

**Comment by DrDiasyl** [comment]
> people will inevitably complain about sec powercreep since humans dont usually have a deathrattle but im of the opinion all species should have one
> 
> though, as i type i realize this IS actually a little sec powercreep because its an audible signal to everyone around that a secoff just died - even if humans deathrattled, it'd still be a very teeny buff
> 
> i dont think its worth really giving a shit about though

Death gasps are blocked by walls. As such, not really a lot if intel you can get (as if you hear it you mostly likely also see it)

File: code/modules/mob/living/emote.dm (hunk 2 above)
Comment (DrDiasyl) [reply code/modules/mob/living/emote.dm:111]: This is a death gasp, if you're using it you're not even unconscious,  you're almost dead.
This is why it counts oxy damage, so make deathgasp play only if you died from any other damage

Comment (nikothedude) [reply code/modules/mob/living/emote.dm:111]: > This is a death gasp, if you're using it you're not even unconscious, you're almost dead. This is why it counts oxy damage, so make deathgasp play only if you died from any other damage

fair, was a bit of a weird question i had anyway

File: code/modules/mob/living/emote.dm [hunk 3]
```diff
@@ -107,9 +107,14 @@
...
+		return //stop the sound if oxyloss too high/cant speak
+	if(istype(carbon_user)) // For masks that give unique death sounds
+		if(isclothing(carbon_user.wear_mask) && (carbon_user.wear_mask.unique_death))
+			playsound(carbon_user, carbon_user.wear_mask.unique_death, 200, TRUE, TRUE)
+			return
 	if(. && user.death_sound)
```
Comment (MrMelbert) [inline code/modules/mob/living/emote.dm:117]: You should be respecting the return value of the parent call, as is being done here. 
IE, just put `!.` in your `can_speak` line.

File: code/modules/mob/living/emote.dm (hunk 2 above)
Comment (MrMelbert) [inline code/modules/mob/living/emote.dm:111]:
```suggestion
	if(!user.can_speak() || user.getOxyLoss() >= 50)
```

**Comment by Helg2** [comment]
To be honest it sounds like something in-between winning the jackpot and dying metrocop

**Comment by carlarctg** [comment]
> To be honest it sounds like something in-between winning the jackpot and dying metrocop

killing a sec off is pretty close to winning the lottery imo

File: code/modules/mob/living/emote.dm (hunk 3 above)
Comment (DrDiasyl) [reply code/modules/mob/living/emote.dm]: I'm kinda confused by this.
Did I do it right now?

**Comment by deathrobotpunch** [comment]
I think you need the end note to last a little bit longer to sound more like a flat line

**Comment by DrDiasyl** [comment]
> I think you need the end note to last a little bit longer with a fade out to sound more like a flat line

It already has a fade out.

**Comment by DrDiasyl** [comment]
@MrMelbert okay I did everything, also checked - it works.

**Comment by DrDiasyl** [comment]
Additional material:

**Comment by carlarctg** [comment]
now give it the death noises from plasma burst 2

File: code/modules/mob/living/emote.dm [hunk 4]
```diff
@@ -107,9 +107,14 @@
...
-	if(. && user.death_sound)
-		if(!user.can_speak() || user.oxyloss >= 50)
-			return //stop the sound if oxyloss too high/cant speak
+	var/mob/living/carbon/carbon_user = user
+	if(!. && !user.can_speak() || user.getOxyLoss() >= 50)
+		return //stop the sound if oxyloss too high/cant speak
+	if(istype(carbon_user)) // For masks that give unique death sounds
+		if(isclothing(carbon_user.wear_mask) && (carbon_user.wear_mask.unique_death))
+			playsound(carbon_user, carbon_user.wear_mask.unique_death, 200, TRUE, TRUE)
+			return
```
Comment (Ghommie) [inline code/modules/mob/living/emote.dm:116]:
```suggestion
	if(!. && !user.can_speak() || user.getOxyLoss() >= 50)
		return //stop the sound if oxyloss too high/cant speak
	var/mob/living/carbon/carbon_user = user
	// For masks that give unique death sounds
	if(istype(carbon_user) && isclothing(carbon_user.wear_mask) && carbon_user.wear_mask.unique_death)
		playsound(carbon_user, carbon_user.wear_mask.unique_death, 200, TRUE, TRUE)
		return
```

**Pull request merged**